### DISCUSSION TITLE
short form of the array assignment

### DISCUSSION THREAD

**Discussion by Artem199405** [nlohmann/json#4511] [Q&A]
Can anyone recommend a short way to represent an array that has sequentially repeated keys for each element? For example, "array": [ 0.0e0, 100(1.0e0),  50(2.0e0), 200(3.0e0), 10(4.0e0), 0.0e0,]

**Comment by nlohmann** [comment]
The question as posted is ambiguous, and the example syntax ('100(1.0e0)') isn't valid JSON or valid nlohmann::json C++ syntax, so it's unclear what 'sequentially repeated keys for each element' of an array is meant to describe -- JSON arrays don't have keys at all (only JSON objects do), so 'repeated keys' doesn't map onto any part of the JSON data model for an array. If the goal is an array whose elements are objects that all share the same set of keys (e.g. [{"value":1.0},{"value":2.0}, ...]), that's just an ordinary array of json objects built the normal way (initializer-list, json::array(), or push_back in a loop) -- there's no separate 'short form' syntax for it. Clarifying the intended JSON output would make it possible to give a concrete example.

---
*This reply was drafted by [Claude Code](https://claude.com/claude-code) on behalf of @nlohmann, as part of a review of open discussions.*
